php_phongo.c

```diff
@@ -2047,6 +2072,16 @@ static mongoc_ssl_opt_t* php_phongo_make_ssl_opt(mongoc_uri_t* uri, zval* zoptio
 	}
 
 #if defined(MONGOC_ENABLE_SSL_SECURE_CHANNEL) || defined(MONGOC_ENABLE_SSL_SECURE_TRANSPORT)
+	if (php_array_existsc(zoptions, "tlsCADirectory")) {
```

Comment:
I based this off of the implementation in #948, not realising that it is wrong. Thank you for the through explanation of what's going on.

Unfortunately, we can't get rid of `php_phongo_make_ssl_opt`, as the `ca_dir`, its deprecated alias `capath`, and `crl_file` options do not have a URI option equivalent. All other driver options now set their canonical `MONGOC_URI_TLS*` parameters in `php_phongo_apply_driver_options_to_uri`. I've adapted `php_phongo_mongoc_ssl_opts_from_uri` to include the same logic as `_mongoc_ssl_opts_from_uri`. However, it would be preferable to have a `mongoc_client_get_ssl_opts` method in the client, as that eliminates the code duplication that we currently have.

> Lastly, if all TLS-related driver options now have proper URI counterparts, I think we can deprecate the URI options in a future release. Feel free to make a ticket for that.

I suppose you meant deprecating the driver options. I've created [PHPC-1430](https://jira.mongodb.org/browse/PHPC-1430) to track adding those deprecations. While looking through this I noted that we don't throw deprecation warnings for other legacy driver options (e.g. `passphrase`). I've added those to the list of deprecations as well.